src/Sylius/Bundle/ShopBundle/Twig/Component/Common/ProductCardComponent.php

```diff
@@ -20,27 +20,52 @@
 use Sylius\Component\Core\Model\ChannelPricingInterface;
 use Sylius\Component\Core\Model\ProductInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+use Sylius\Component\Locale\Context\LocaleContextInterface;
 use Sylius\Component\Product\Resolver\ProductVariantResolverInterface;
 use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
 use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+use Symfony\UX\TwigComponent\Attribute\PostMount;
 
 #[AsTwigComponent]
 class ProductCardComponent
 {
-    public ProductInterface $product;
+    public ?ProductInterface $product = null;
 
-    public ProductVariantInterface $variant;
+    public ?ProductVariantInterface $variant = null;
 
+    public ?string $slug = null;
+
+    /**
+     * @param ProductRepositoryInterface<ProductInterface> $productRepository
+     */
     public function __construct(
+        private readonly ProductRepositoryInterface $productRepository,
         private readonly ProductVariantResolverInterface $productVariantResolver,
         private readonly ChannelContextInterface $channelContext,
+        private readonly LocaleContextInterface $localeContext,
         private readonly ProductVariantPricesCalculatorInterface $productVariantPricesCalculator,
     ) {
     }
 
+    #[PostMount]
+    public function postMount(): void
+    {
+        if ($this->product === null && $this->slug !== null) {
+            $this->setProductBySlug($this->slug);
+        }
+    }
+
+    #[ExposeInTemplate(name: 'product')]
+    public function getProduct(): ?ProductInterface
+    {
+        return $this->product;
+    }
```

Comment:
```suggestion
```

Reply:
it's not gonna work if we are setting product by slug, forgot to unresolve the comment above

Reply:
but we just need to expose it, so removed the method and just added the attribute to product